src/clj/game/cards/ice.clj

```diff
@@ -1583,6 +1583,70 @@
    "Rototurret"
    {:subroutines [trash-program end-the-run]}
 
+   "Sadaka"
+   {:subroutines [{:label "Look at the top 3 cards of R&D and either arrange them in any order or shuffle R&D. You may draw 1 card."
+                   :req (req (not-empty (:deck corp)))
+                   :delayed-completion true
+                   :effect (req (let [c (take 3 (:deck corp))
+                                      c-names (map :title c)
+                                      maybe-draw (fn [state side card]
```

Comment:
I would honestly move this function into a `let` outside the card map. Don't make it a function, make it an ability map with an effect that starts with the next line. That ability will be delayed completion, as will the optional's yes-ability, because `draw` is awaitable due to interactions with Jinja City Grid.